CONFIGURATION.md

```diff
@@ -0,0 +1,58 @@
+# Configuration and common usecases
+## Tests are run in upstream environment
+
+Pytest will run local candlepin service in a container.
+
+Dynaconf will load all configuration from a file `settings.toml`
+
+```bash
+$ export ENVIRONMENTS_FOR_DYNACONF=True
+$ export ENV_FOR_DYNACONF=upstream
+$
+$ pytest -v
+```
+Pytest will run all tests that are available for `upstream`
+environment.
+
+There is a section `[upstream]` in the settings file that contains of
+all properties that tests require.
+
+## Tests are run in CI environment
+Jenkins does not require dynaconf to use environments.
+Jenkins provides all config properties directly - using environment
+variables.
+
+```bash
+$ export ENVIRONMENTS_FOR_DYNACONF=False
+$
+$ pytest -v
+```
+
+## Tests are run in stage environment
+Properties that describe a development environment must be stored in a
+secret file that is not part of the repo. Default name of the file is
+`.secrets.toml`
+
+```bash
+$ export ENVIRONMENTS_FOR_DYNACONF=True
+$ export ENV_FOR_DYNACONF=stage
+$
+$ pytest -v
+```
+
+`stage` environment is used for most of the tests. It requires all
+services to be available.
+
+All properties are defined in a custom file `.secrets.toml` This file
+is not part of this repo.
```

Comment:
There is difference that upstream environment will be already configured in the repo (ie. all config properties will be in a file settings.toml (or generated automatically from test candlepin). So the most important part of this section is note about .secrets.toml that is not part of the repo and it is necessary to create it.

Reply:
The only difference I see is related to the configuration file for dynaconf; since that will need to be explained/documented regardless of the environment used, then the two different sections become basically redundant.

Please un-duplicate them, thanks.